swarm/src/behaviour.rs

```diff
@@ -101,11 +101,57 @@ pub trait NetworkBehaviour: Send + 'static {
     fn inject_connection_closed(&mut self, _: &PeerId, _: &ConnectionId, _: &ConnectedPoint)
     {}
 
-    /// Informs the behaviour about an event generated by the handler dedicated to the peer identified by `peer_id`.
-    /// for the behaviour.
+    /// Informs the behaviour that one or more connections to a peer are
+    /// currently busy and unable to receive the given event which was
+    /// previously emitted by `NetworkBehaviour::poll`.
```

Comment:
```suggestion
    /// emitted by the last call to `NetworkBehaviour::poll`.
```

How about documenting and thus guaranteeing that the event passed to `inject_connection_busy` is always the last event received from `NetworkBehaviour::poll` and not any of the previous ones from the past.

Reply:
What would be the point of doing that?

Reply:
A `NetworkBehaviour` can thus be sure that all previously emitted events via `NetworkBehaviour::poll` have been handled and won't be passed back via `inject_connection_busy` out-of-order at a later point in time.

Reply:
> A `NetworkBehaviour` can thus be sure that all previously emitted events via `NetworkBehaviour::poll` have been handled and won't be passed back via `inject_connection_busy` out-of-order at a later point in time.

Naturally, but my question was more aiming at: How is it useful for the behaviour to know that it is always the last event emitted from `poll`? Note that the `NetworkBehaviour` already guarantees that all events emitted for the connection of a peer are delivered in the same order to the handler as they are emitted. That implies that if the behaviour gets an event back via `inject_connections_busy` all previous events have either been delivered or also given to `inject_connections_busy`. This is already very close to the guarantee you want to give, but avoids being as specific as referring to the "last call to poll", which I think is an unnecessary over-specification. The ordering guarantees w.r.t. emitted events could maybe be made clearer in the docs, what do you think?

Reply:
> The ordering guarantees w.r.t. emitted events could maybe be made clearer in the docs, what do you think?

Yes, with the already existing ordering guarantees in mind my initial suggestion is superfluous. Where are these documented?